lib/understream.coffee

```diff
@@ -92,8 +94,12 @@ class ArrayStream extends Readable
   constructor: (@options, @arr, @index=0) ->
     super _(@options).extend objectMode: true
   _read: (size) =>
-    debug "_read #{size} #{JSON.stringify @arr[@index]}"
-    @push @arr[@index] # Note: push(undefined) signals the end of the stream, so this just works^tm
+    data = @arr[@index]
+    if data is undefined
```

Comment:
Another option would be to push `null` if `@index` > `@arr.length - 1` and that would allow reading `undefined` values from the stream. I'm not sure if we'd actually want that though.

Reply:
Yeah I'm not sure either 🤔

Reply:
I think we should.. adding that change